File: src/vibe_kanban_clone/schemas/task_comment.py
"""TaskComment schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class TaskCommentCreate(BaseModel):
    """Schema for creating a task comment."""

    author: str
    body: str


class TaskCommentRead(BaseModel):
    """Schema for reading a task comment."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    task_id: int
    author: str
    body: str
    created_at: datetime

File: server/src/schemas/taskComment.ts
import { z } from "zod";

export const TaskCommentCreate = z.object({
  author: z.string().min(1),
  body: z.string().min(1),
});

export const TaskCommentRead = z.object({
  id: z.number(),
  task_id: z.number(),
  author: z.string(),
  body: z.string(),
  created_at: z.string().or(z.date()),
});

export type TaskCommentCreate = z.infer<typeof TaskCommentCreate>;
export type TaskCommentRead = z.infer<typeof TaskCommentRead>;
